src/services/BavService.ts

```diff
@@ -0,0 +1,180 @@
+import { SendMessageCommand } from "@aws-sdk/client-sqs";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { DynamoDBDocument, GetCommand, PutCommand } from "@aws-sdk/lib-dynamodb";
+import { randomUUID } from "crypto";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { ISessionItem } from "../models/ISessionItem";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { SharedClaimsPersonIdentity, PersonIdentityItem, PersonIdentityName, PersonIdentityDateOfBirth } from "../models/PersonIdentityItem";
+import { AppError } from "../utils/AppError";
+import { absoluteTimeNow } from "../utils/DateTimeUtils";
+import { sqsClient } from "../utils/SqsClient";
+import { TxmaEvent } from "../utils/TxmaEvent";
+
+export class BavService {
+	readonly tableName: string;
+
+	private readonly dynamo: DynamoDBDocument;
+
+	readonly logger: Logger;
+
+	private static instance: BavService;
+
+	constructor(tableName: string, logger: Logger, dynamoDbClient: DynamoDBDocument) {
+		this.tableName = tableName;
+		this.dynamo = dynamoDbClient;
+		this.logger = logger;
+	}
+
+	static getInstance(tableName: string, logger: Logger, dynamoDbClient: DynamoDBDocument): BavService {
+		if (!BavService.instance) {
+			BavService.instance = new BavService(tableName, logger, dynamoDbClient);
+		}
+		return BavService.instance;
+	}
+
+	async getSessionById(sessionId: string): Promise<ISessionItem | undefined> {
+		this.logger.debug("Table name " + this.tableName);
+		const getSessionCommand = new GetCommand({
+			TableName: this.tableName,
+			Key: {
+				sessionId,
+			},
+		});
+
+		let session;
+		try {
+			session = await this.dynamo.send(getSessionCommand);
+		} catch (error) {
+			this.logger.error({ message: "getSessionById - failed executing get from dynamodb:" }, {
+				messageCode: MessageCodes.FAILED_FETCHING_SESSION,
+				error,
+			});
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error retrieving Session");
+		}
+
+		if (session.Item) {
+			if (session.Item.expiryDate < absoluteTimeNow()) {
```

Comment:
If we throw this under line 47 within the try will the catch also not "catch" if session.Item is null?

Reply:
Sorry not sure what you mean here - do you want the expired error to be caught by the try/catch on 47?